typescript-decorators/src/decorators.ts

```diff
@@ -0,0 +1,31 @@
+import { FastifyReply, FastifyRequest } from 'fastify'
+import { IncomingMessage, ServerResponse } from 'http'
+import { instance } from './instance'
+
+export interface RouteConfig {
+    url: string
+}
+
+// Interface which should implement each handler
+export interface RequestHandler {
+    request: FastifyRequest<IncomingMessage>
+    response: FastifyReply<ServerResponse>
+
+    handle(): void | Promise<any>
+}
+
+// Typescript decorator implementation
+export function GET(config: RouteConfig) {
+    return (constructor: any) => {
+        /**
+         * Magic happens here
```

Comment:
I prefer we do not use the term "magic". We need to explain magic.

Reply:
Magic explained below. It works because of node require.cache -> everything we import instance we import the same object.

If description doesn't clear enough then I will rewrite it